internal/certs/certs_test.go

```diff
@@ -0,0 +1,118 @@
+package certs
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("CertManager", func() {
+	var (
+		dir string
+		cm  CertManager
+	)
+	BeforeEach(func() {
+		var err error
+		dir, err = os.MkdirTemp("", "certs")
+		Expect(err).NotTo(HaveOccurred())
+		cm = CertManager{CertificatesDir: dir}
+	})
+
+	AfterEach(func() {
+		Expect(os.RemoveAll(dir)).To(Succeed())
+	})
+
+	It("generateCA success", func() {
+		certPath := filepath.Join(dir, "admin-kubeconfig-client-ca.crt")
+		keyPath := filepath.Join(dir, "admin-kubeconfig-client-ca.key")
+		adminCA := CertInfo{
+			commonName:      "admin-kubeconfig-signer",
+			certificatePath: certPath,
+			keyPath:         keyPath,
+		}
+		_, err := generateAndWriteCA(adminCA)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = os.Stat(certPath)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = os.Stat(keyPath)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("generateSignerCerts success", func() {
+		err := cm.generateSignerCerts()
+		Expect(err).NotTo(HaveOccurred())
+		// verify all signer keys exists
+		signerKeyFileNames := []string{
+			"loadbalancer-serving-signer.key",
+			"localhost-serving-signer.key",
+			"service-network-serving-signer.key",
+			"ingresskey-ingress-operator.key",
+		}
+		for _, fileName := range signerKeyFileNames {
+			// verify all signer keys exists
+			_, err = os.Stat(filepath.Join(dir, fileName))
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	It("generateAdminUserCertificate success", func() {
+		certPath := filepath.Join(dir, "admin-kubeconfig-client-ca.crt")
+		keyPath := filepath.Join(dir, "admin-kubeconfig-client-ca.key")
+		adminCA := CertInfo{
+			commonName:      "admin-kubeconfig-signer",
+			certificatePath: certPath,
+			keyPath:         keyPath,
+		}
+		ca, err := generateAndWriteCA(adminCA)
+		caCert := loadCACertFromFile(certPath)
+
+		userCert, _, err := generateAdminUserCertificate(ca)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Verify that the client cert was signed by the given
```

Comment:
Was there supposed to be more to this comment? Given what?

Reply:
Yes, I added the verification in line 83